aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,327 @@
+
+.. note:: ``IOrderedFilter`` trumps scope when determining the order in which filters will run. Filters are sorted first by order, then scope is used to break ties. Order defaults to 0 if not set.
+
+To modify the default, scope-based order, you could explicitly set the ``Order`` property of the class-level and/or method-level filters. For example, adding ``Order=-1`` to the action method:
+
+.. code-block:: c#
+
+	[OrderLoggingActionFilter(Name = "Method Level Attribute", Order=-1)]
+
+In this case, a value of less than zero would ensure this filter ran before both the Global and Class level filters.
+
+The new order would be:
+	#. The Controller ``OnActionExecuting``
+	#. The Method filter ``OnActionExecuting``
+	#. The Global filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuted``
+	#. The Global filter ``OnActionExecuted``
+	#. The Method filter ``OnActionExecuted``
+	#. The Controller ``OnActionExecuted``
+
+.. note:: The ``Controller`` class's methods always run before and after all filters. These methods are not implemented as ``IFilter`` instances and do not participate in the ``IFilter`` ordering algorithm.
+
+.. _authorization-filters:
+
+Authorization Filters
+---------------------
+*Authorization Filters* control access to action methods, and are the first filters to be executed within the filter pipeline. They have only a before stage, unlike most filters that support before and after methods. You should only write a custom authorization filter if you are writing your own authorization framework. Note that you should not throw exceptions within authorization filters, since nothing will handle the exception (exception filters won't handle them). Instead, issue a challenge or find another way. They are covered in the :doc:`Security </security/index>` section of the documentation.
+
+Learn more about :doc:`/security/authorization/authorization-filters`.
+
+.. _resource-filters:
+
+Resource Filters
+----------------
+*Resource Filters* implement either the ``IResourceFilter`` or ``IAsyncResourceFilter`` interface, and their execution wraps most of the filter pipeline (only :ref:`authorization-filters` run before them - all other filters and action processing happens between their ``OnResourceExecuting`` and ``OnResourceExecuted`` methods). Resource filters are especially useful if you need to short-circuit most of the work a request is doing. Caching would be one example use case for a resource filter, since if the response is already in the cache, the filter can immediately set a result and avoid the rest of the processing for the action.
+
+The :ref:`short circuiting resource filter <short-circuiting-resource-filter>` shown above is one example of a resource filter. A very naive cache implementation (do not use this in production) is shown below:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/NaiveCacheResourceFilterAttribute.cs
```

Comment:
Depends what action result it is. It might be better to just show caching a string and using `new ContentResult(...)`